pancloud/adapters/adapter.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+
+"""Base adapter class."""
+from __future__ import absolute_import
+
+from abc import ABCMeta, abstractmethod
+import six
+
+
+@six.add_metaclass(ABCMeta)
```

Comment:
not a huge deal, but I prefer to not have six as a dependency and handle the py2/py3 differences natively.  I think there are some small conditionals needed the ABC case.  if not or they suck six is ok.

Reply:
I was torn here. I even contemplated using a `raise NotImplementedError` pattern instead of an ABC. I'm ok using conditionals so long as the pattern is clean and concise. Otherwise, we can use `six` until we decide to retire support for Python 2.